src/api/event-services/work-request.js

```diff
@@ -40,6 +78,12 @@ const handleWorkRequest = async (workRequest, socket) => {
 
       const workSubmitService = new WorkSubmitService(workRequest);
       await workSubmitService.submitWork();
+    } else if (e instanceof WorkRequestError) {
+      logger.log(e.message);
+      logger.log('Work request error : ', e.message);
+
+      socket.emit(`WorkResponse-${uuid}`, createWorkResponseError(e.message));
+      logger.log(`Error response sent back to ${uuid}`);
```

Comment:
I am pretty sure in the handler for `WorkRequest` socket events there is automatically a system for catching all errors and reporting them in a certain way. No need for custom error types.

Reply:
see e.g. here: https://github.com/biomage-ltd/api/blob/develop/src/api/events/index.js#L39-L52

Reply:
Hmm.. How about if I just rethrow the error?